.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - support/*
+  workflow_dispatch:
+  workflow_call:
+
+env:
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+  DOTNET_NOLOGO: true
+  DOTNET_GENERATE_ASPNET_CERTIFICATE: false
+
+jobs:
+
+  build-windows:
+    name: Build on Windows
+    runs-on: windows-2022
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: Run build script
+        run: ./build.cmd
+        shell: cmd
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: windows-artifacts
+          path: artifacts\**\*
+
+  build-ubuntu:
+    name: Build on Ubuntu
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: |
+            5.0
+            3.1
+            2.1
+      - name: Run build script
+        run: ./build.sh
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: ubuntu-artifacts
+          path: artifacts/**/*

```

Comment:
Nothing wrong with this, but it's a shame we can't use a matrix and avoid all the duplication. But we'd need to deal with the different build commands as well as the need for extra runtimes on Linux but not Windows.
I wouldn't make any changes. Just making conversation.

Reply:
As pleases you. If you think it's fun, go ahead. Otherwise, just carry on. I have no problem with what you have.

(although, does `./build.ps1` work on both OSes?)